jsonschema_extractor/attrs_extractor.py

```diff
@@ -37,30 +38,47 @@ def extract(cls, extractor, typ):
     def _extract_attribute(cls, extractor, attribute):
         is_required = attribute.default is attr.NOTHING
 
-        schema = None
         if "jsonschema" in attribute.metadata:
             schema = attribute.metadata["jsonschema"]
-        elif attribute.type is not None:
-            schema = extractor.extract(attribute.type)
         else:
+            schema = cls._extract_attribute_schema_by_type(extractor, attribute) or {}
             for validator in _iterate_validator(attribute.validator):
-                if isinstance(validator, _InstanceOfValidator):
-                    schema = extractor.extract(validator.type)
+                if isinstance(validator, _InValidator):
+                    schema['enum'] = validator.options
 
-        if schema is None:
+        if not schema:
```

Comment:
should this check live above the "for validator in.." call? I the schema could be None here.

Reply:
It cannot be `None` because in `attrs_extractor:44` add Added `or {}` at the end.  
I didn't want to insert the `InValidator` logic into `_extract_attribute_schema_by_type` function as it should only handle extraction by type.  
In case you prefer a more explicit `if` inside the `for validator in ...` I can change it, however I think that this is the more elegant solution.

Reply:
Nevermind, I see why that it can't be None. 

And yes, I think it would probably be better to move the enum extraction logic to the _extract_attribute function, since the enum field is part of the schema anyway.

But it's a quick patch I could do afterward.